core/client/src/runtime_api.rs

```diff
@@ -100,8 +99,10 @@ decl_runtime_apis! {
 
 	/// The `Metadata` api trait that returns metadata for the runtime.
 	pub trait Metadata {
+		/// Deprecated version of metadata for runtime
+		fn metadata_old() -> OpaqueMetadata;
 		/// Returns the metadata of a runtime.
-		fn metadata() -> OpaqueMetadata;
+		fn metadata(version: u16) -> OpaqueMetadata;
```

Comment:
We should always return the latest version, we don't want to maintain multiple encoders.
The version should give just the caller a chance to decode without failing.

Reply:
Couldn't a few deprecated version be maintained? (in fact changing it just require removing the parameter as it is the default behavior).

Reply:
No, that just increases the size of the wasm blob. The ui will just be able to show a nice error messsage "Please update" instead of just breaking silently.